File: Front-End/src/services/category.service.ts

import axiosClient from '@/configurations/axios.config';
import type { 
  CreateCategoryRequest, 
  CategoryResponse, 
  CategoryListResponse 
} from '@/types/category.type';

export const categoryService = {
  getCategories: async (page: number = 1, size: number = 10, search: string = "") => {
    const response = await axiosClient.get<CategoryListResponse>(`/categories?page=${page}&size=${size}&categoryName=${search}`);
    return response.data;
  },

  getCategoryById: async (id: number) => {
    const response = await axiosClient.get<CategoryResponse>(`/categories/${id}`);
    return response.data;
  },

  createCategory: async (request: CreateCategoryRequest) => {
    const response = await axiosClient.post<CategoryResponse>('/categories', request);
    return response.data;
  },

  updateCategory: async (id: number, request: CreateCategoryRequest) => {
    const response = await axiosClient.put<CategoryResponse>(`/categories/${id}`, request);
    return response.data;
  },

  changeStatusCategory: async (id: number) => {
    await axiosClient.put(`/categories/change-status/${id}`);
  }
};

File: Back-End/src/main/java/iuh/fit/ecommerce/services/CategoryService.java
package iuh.fit.ecommerce.services;

import iuh.fit.ecommerce.entities.Category;
import jakarta.validation.Valid;
import iuh.fit.ecommerce.dtos.request.category.CategoryAddRequest;
import iuh.fit.ecommerce.dtos.response.base.ResponseWithPagination;
import iuh.fit.ecommerce.dtos.response.category.CategoryResponse;

import java.util.List;

public interface CategoryService {

    CategoryResponse createCategory(CategoryAddRequest request);

    ResponseWithPagination<List<CategoryResponse>> getCategories( int page, int size, String categoryName);

    CategoryResponse getCategoryById(Long id);

    CategoryResponse updateCategory(Long id, @Valid CategoryAddRequest request);

    void changeStatusCategory(Long id);
    Category getCategoryEntityById(Long id);
}
